lib/gosu.rb

```diff
@@ -17,10 +17,369 @@
+  attach_function :Gosu_axis,              [:uint32], :double
+  attach_function :Gosu_button_id_to_char, [:uint32], :string
+  attach_function :Gosu_button_char_to_id, [:string], :uint32
+  attach_function :Gosu_button_name,       [:uint32], :string
+  attach_function :Gosu_gamepad_name,      [:uint32], :string
+
+  attach_function :Gosu_clipboard,      [], :string
+  attach_function :Gosu_set_clipboard,  [:string], :void
+
+  attach_function :Gosu_draw_line,     [:double, :double, :uint32, :double, :double, :uint32, :double, :uint32], :void
+  attach_function :Gosu_draw_triangle, [:double, :double, :uint32, :double, :double, :uint32,
+                                        :double, :double, :uint32, :double, :uint32],                            :void
+  attach_function :Gosu_draw_quad,     [:double, :double, :uint32, :double, :double, :uint32,
+                                        :double, :double, :uint32, :double, :double, :uint32, :double, :uint32], :void
+  attach_function :Gosu_draw_rect,     [:double, :double, :double, :double, :uint32, :double, :uint32],          :void
+
+  attach_function :Gosu_offset_x,   [:double, :double],                   :double
+  attach_function :Gosu_offset_y,   [:double, :double],                   :double
+  attach_function :Gosu_distance,   [:double, :double, :double, :double], :double
+  attach_function :Gosu_angle,      [:double, :double, :double, :double], :double
+  attach_function :Gosu_angle_diff, [:double, :double],                   :double
+  attach_function :Gosu_random,     [:double, :double],                   :double
+
+  attach_function :Gosu_screen_width,     [:pointer], :uint32
+  attach_function :Gosu_screen_height,    [:pointer], :uint32
+  attach_function :Gosu_available_width,  [:pointer], :uint32
+  attach_function :Gosu_available_height, [:pointer], :uint32
+
+  attach_function :Gosu_last_error, [], :string
+
+  def self.fps
+    Gosu_fps().tap { Gosu.check_last_error }
+  end
+
+  def self.flush
+    Gosu_flush()
+    Gosu.check_last_error
+  end
+
+  def self.milliseconds
+    Gosu_milliseconds().tap { Gosu.check_last_error }
+  end
+
+  def self.default_font_name
+    Gosu_default_font_name().tap { Gosu.check_last_error }
+  end
+
+  def self.user_languages
+    languages = []
+    callback = proc { |data, string| languages << string if string }
+    Gosu_user_languages(callback, nil)
+    Gosu.check_last_error
+    languages
+  end
+
+  def self.gl(z = nil, &block)
+    # Gosu.gl might not be called immediately
+    # so to prevent block getting GC'd until used
+    # store them in a global variable which is
+    # cleared after rendering is complete
+    # i.e. after Window.draw is done.
+    $gosu_gl_blocks ||= []
+    $gosu_gl_blocks << block
+
+    raise "Block not given!" unless block
+
+    if z
+      Gosu_gl_z(z, block)
+      Gosu.check_last_error
+    else
+      Gosu_gl(block)
+      Gosu.check_last_error
+    end
```

Comment:
```suggestion
    z ? Gosu_gl_z(z, block) :  Gosu_gl(block)
    Gosu.check_last_error
```

Reply:
I still went with `if else end` because I prefer to use ?: only for expressions, but I have DRY'd the check_last_error call, thanks.